File: code_wars/8kyu/alternate/alternate.cpp
#include <string>
#include <cctype>
using namespace std;

std::string to_alternating_case(const std::string &str)
{
    std::string result{};
    for (auto item : str)
    {
        if (islower(item))
            result += ((char)toupper(item));
        else
            result += ((char)tolower(item));
    }
    return result;
}

File: code_wars/8kyu/alternate/alternate.go
package kata

import (
	"strings"
	"unicode"
)

func ToAlternatingCase(str string) string {
	result := ""
	for _, i := range str {
		if unicode.IsUpper(i) && unicode.IsLetter(i) {
			result += strings.ToLower(string(i))
		} else if unicode.IsLower(i) && unicode.IsLetter(i) {
			result += strings.ToUpper(string(i))
		} else {
			result += string(i)
		}
	}
	return result
}
